File: llm_generated_code/ChatGPT/CPP/Level2/Q17.cpp
#include <iostream>
#include <vector>
using namespace std;

int maxTreasureValue(const vector<int>& treasures) {
    int maxSum = treasures[0];
    int currentSum = treasures[0];

    for (size_t i = 1; i < treasures.size(); ++i) {
        currentSum = max(treasures[i], currentSum + treasures[i]);
        maxSum = max(maxSum, currentSum);
    }

    return maxSum;
}

int main() {
    vector<int> treasures = {3, -2, 5, -1, 4, -3, 2, 1};
    cout << "Maximum treasure value: " << maxTreasureValue(treasures) << endl;
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level2/Q17.py
def max_treasure_value(map_segments):
    max_sum = current_sum = map_segments[0]
    for value in map_segments[1:]:
        current_sum = max(value, current_sum + value)
        max_sum = max(max_sum, current_sum)
    return max_sum

map_segments = [3, -2, 5, -1, 4, -3, 2, 1]
print(max_treasure_value(map_segments))
